src/rp2_common/pico_cyw43_arch/cyw43_arch.c

```diff
@@ -123,6 +123,24 @@ void cyw43_hal_get_mac(__unused int idx, uint8_t buf[6]) {
     memcpy(buf, cyw43_state.mac, 6);
 }
 
+// Return some characters from the ascii representation of the mac address
+// e.g. 112233445566
+// chr_off is index of character in mac to start
+// chr_len is length of result
+// chr_off=8 and chr_len=4 would return "5566"
+// Return number of characters put into destination
+size_t cyw43_hal_get_mac_ascii(int idx, size_t chr_off, size_t chr_len, char *dest_in) {
+    static const char hexchr[16] = "0123456789ABCDEF";
+    uint8_t mac[6];
+    char *dest = dest_in;
+    assert(chr_len <= (2 * sizeof(mac)) && chr_off + chr_len <= (2 * sizeof(mac)));
```

Comment:
Not that it really matters, but I guess if `chr_off + chr_len <= (2 * sizeof(mac))` then `chr_len <= (2 * sizeof(mac))` is implicitly true? :wink:

Reply:
yes - let me get rid of it